code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+	if(!chosen_name)
+		to_chat(src, "<span class='warning'>Not a valid name, please try again.</span>")
+		dragon_name()
+		return
+	to_chat(src, "<span class='notice'>Your name is now <span class='name'>[chosen_name]</span>, the feared Space Dragon.</span>")
+	fully_replace_character_name(null, chosen_name)
+
+/**
+  * Allows space dragon to choose a color for itself.
+  *
+  * Prompts the space dragon to choose a color, from which it will then apply to itself.
+  * If an invalid color is given, will re-prompt the dragon until a proper color is chosen.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/color_selection()
+	chosen_color = input(src,"What would you like your color to be?","Choose Your Color", COLOR_WHITE) as color|null
+	if(!chosen_color) //redo proc until we get a color
+		to_chat(src, "<span class='warning'>Not a valid color, please try again.</span>")
+		color_selection()
+		return
+	var/temp_hsv = RGBtoHSV(chosen_color)
+	if(ReadHSV(temp_hsv)[3] < DARKNESS_THRESHOLD)
+		to_chat(src, "<span class='danger'>Invalid color. Your color is not bright enough.</span>")
+		color_selection()
+		return
+	add_atom_colour(chosen_color, FIXED_COLOUR_PRIORITY)
+	add_dragon_overlay()
+
+/**
+  * Adds the proper overlay to the space dragon.
+  *
+  * Clears the current overlay on space dragon and adds a proper one for whatever animation he's in.
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/add_dragon_overlay()
+	cut_overlays()
+	if(!small_sprite.small)
+		return
+	if(stat == DEAD)
+		var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_dead")
+		overlay.appearance_flags = RESET_COLOR
+		add_overlay(overlay)
+		return
+	if(!using_special)
+		var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_base")
+		overlay.appearance_flags = RESET_COLOR
+		add_overlay(overlay)
+		return
+	if(using_special)
+		var/mutable_appearance/overlay = mutable_appearance(icon, "overlay_gust")
+		overlay.appearance_flags = RESET_COLOR
+		add_overlay(overlay)
+
+/**
+  * Determines a line of turfs from sources's position to the target with length range.
+  *
+  * Determines a line of turfs from the source's position to the target with length range.
+  * The line will extend on past the target if the range is large enough, and not reach the target if range is small enough.
+  * Arguments:
+  * * offset - whether or not to aim slightly to the left or right of the target
+  * * range - how many turfs should we go out for
+  * * atom/at - The target
+  */
+/mob/living/simple_animal/hostile/space_dragon/proc/line_target(offset, range, atom/at = target)
+	if(!at)
+		return
+	var/angle = ATAN2(at.x - src.x, at.y - src.y) + offset
+	var/turf/T = get_turf(src)
+	for(var/i in 1 to range)
+		var/turf/check = locate(src.x + cos(angle) * i, src.y + sin(angle) * i, src.z)
+		if(!check)
+			break
+		T = check
```

Comment:
Whats the point of this check?

Reply:
If its to find the point that is range units away from the dragon, just use some basic trig to find it directly without needing to loop

Reply:
After reading the method doc I still have a very vague idea, but it's not for getting a specific spot x units away, it's for... creating a line of turfs? As an array? It's weird.